test/unit/algo/algorithm/sums_special_cases.cpp

```diff
@@ -69,21 +66,57 @@ TTS_CASE("eve.algo.inclusive_scan complex numbers")
   std::vector<float> real_copy = real;
   std::vector<float> img_copy  = img;
 
-  kumi::tuple<float, float> init{0.0, 0.0};
+  kumi::tuple<float, float> init {0.0, 0.0};
 
-  eve::algo::inclusive_scan_to(
-    eve::algo::views::zip(real, img),
-    eve::algo::views::zip(real_copy, img_copy),
-    std::pair{plus, eve::zero},
-    init);
+  eve::algo::inclusive_scan_to(eve::algo::views::zip(real, img),
+                               eve::algo::views::zip(real_copy, img_copy),
+                               std::pair {plus, eve::zero},
+                               init);
 
-  eve::algo::inclusive_scan_inplace(eve::algo::views::zip(real, img), std::pair{plus, eve::zero}, init);
+  eve::algo::inclusive_scan_inplace(
+      eve::algo::views::zip(real, img), std::pair {plus, eve::zero}, init);
 
-  std::vector<float> expected_real = { 0.0,  0.1,  0.3,  0.6 };
-  std::vector<float> expected_img  = { 0.0, -0.1, -0.3, -0.6 };
+  std::vector<float> expected_real = {0.0, 0.1, 0.3, 0.6};
+  std::vector<float> expected_img  = {0.0, -0.1, -0.3, -0.6};
   TTS_EQUAL(expected_real, real);
-  TTS_EQUAL(expected_img,  img);
+  TTS_EQUAL(expected_img, img);
 
   TTS_EQUAL(expected_real, real_copy);
-  TTS_EQUAL(expected_img,  img_copy);
+  TTS_EQUAL(expected_img, img_copy);
+};
+
+TTS_CASE("eve.algo.transform_reduce_simple")
```

Comment:
or should the stateful reduction happen here?

Reply:
I don't know a good example on stateful reduction. Stateful map added.